packages/commonwealth/server/routes/getNewProfile.ts

```diff
@@ -0,0 +1,92 @@
+import type { Request, Response, NextFunction } from 'express';
+import { Op } from 'sequelize';
+import type { DB } from '../models';
+
+export const Errors = {
+  NoAddressProvided: 'No address provided in query',
+  NoAddressFound: 'No address found',
+  NoProfileFound: 'No profile found',
+};
+
+const getNewProfile = async (
+  models: DB,
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  const { address } = req.query;
+  if (!address) return next(new Error(Errors.NoAddressProvided));
+
+  const addressModel = await models.Address.findOne({
+    where: {
+      address,
+    },
+    include: [models.Profile],
+  });
+  if (!addressModel) return next(new Error(Errors.NoAddressFound));
+
+  const profile = await models.Profile.findOne({
+    where: {
+      id: addressModel.profile_id,
+    },
+  });
+  if (!profile) return next(new Error(Errors.NoProfileFound));
+
+  const addresses = await models.Address.findAll({
+    where: {
+      profile_id: profile.id,
+    },
+  });
+
+  const chainIds = [...new Set<string>(addresses.map((a) => a.chain))];
+  const chains = await models.Chain.findAll({
+    where: {
+      id: {
+        [Op.in]: chainIds,
+      },
+    },
+  });
+
+  const addressIds = [...new Set<number>(addresses.map((a) => a.id))];
+  const threads = await models.Thread.findAll({
```

Comment:
Threads and Comments could also be handled via `include` on the query for Addresses.

Reply:
Discussed and because this ties in with collaborations, will leave as is for now.